### DISCUSSION TITLE
Interrupt Handling with Custom Plugins

### DISCUSSION THREAD

**Discussion by AdamWHuffman** [avocado-framework/avocado#5872] [Q&A]
I made a pre/post test runner plugin so that my VM hosts are created with ansible.
How can I automatically run the post test runners when I interrupt the process.

For example
```
avocado run mytestscript -m <yaml config>
CTRL+c
```

In the interrupt case, my assets are left created, but if the test finishes, they are cleaned up.

**Comment by richtja** [comment]
Hi @AdamWHuffman if your `mytestscript` is avocado-instrumented test, you can use TearDown method for any clean-ups, but if you need to use plugins for that, we don't support such behaviour right now. 

We can discuss implementation of such feature in #5871
